tt_metal/llrt/rtoptions.cpp

```diff
@@ -56,6 +57,10 @@ RunTimeOptions::RunTimeOptions() {
         if (profile_dispatch_str != nullptr && profile_dispatch_str[0] == '1') {
             profile_dispatch_cores = true;
         }
+        const char *profiler_sync_enabled_str = std::getenv("TT_METAL_PROFILER_SYNC");
```

Comment:
Man another env var 🥴 
@TT-billteng

Reply:
These env vars will be hidden behind `tracy.py` cli options pretty soon. User never need to now about them. I need [this](https://github.com/tenstorrent/tt-metal/issues/7116) ticket in before I can add more CLI options.

Reply:
can you pass this via function parameters somehow?

Reply:
It is a runtime variable. We are deciding at runtime.

Reply:
you can pass runtime variables from the top all the way down through the stack?

Reply:
In tracy.py I can bring tt-lib bindings for this switch and set it through that. That way I can avoid the env_var